### PR TITLE
Add AdultFilmIndex parser

### PR DESCRIPTION
**Pull request by themese8**
Hello, we are building a new adult filmography database on AdultFilmIndex.com. We combine the metadata from multiple adult film sources to make a new, comprehensive adult DVD index and database.

We have made a stash-specific API endpoints for scraping the metadata for movies and scenes. The API is in beta, and we have included a key for accessing it, this key is available for all stash users.

This PR includes the parsers for scenes and movies using AdultFilmIndex API. We have tested the parser locally and have determined it working.

### CODE REVIEW

File: scrapers/AdultFilmIndex.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  sceneScraper:
+    scene:
+      Title: data.title
+      Details: data.description
+      URL: data.url
+      Image: data.thumbnail
+      Date:
+        selector: data.movie.release_date
+        postProcess:
+          - parseDate: 2006-01-02
```
Comment (bnkai) [inline scrapers/AdultFilmIndex.yml:46]: it seems that we need some pp to remove the `T.*` part of the date
```suggestion
          - replace:
              - regex: T.*$
                with: ""
          - parseDate: 2006-01-02
```

File: scrapers/AdultFilmIndex.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  movieScraper:
+    movie:
+      Name: data.title
+      Synopsis: data.description
+      URL: data.url
+      Duration: data.runtime
+      Date:
+        selector: data.release_date
+        postProcess:
+          - parseDate: 2006-01-02
```
Comment (bnkai) [inline scrapers/AdultFilmIndex.yml:64]: same as above
```suggestion
          - replace:
              - regex: T.*$
                with: ""
          - parseDate: 2006-01-02
```

**Comment by themese8** [comment]
Thanks for the review, updated the PR

File: scrapers/AdultFilmIndex.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  scraper: sceneScraper
+movieByURL:
+  - action: scrapeJson
+    url:
+      - https://adultfilmindex.com/movie/
+    scraper: movieScraper
+    queryURL: https://adultfilmindex.com/api/v1/stash/movie/{url}
+    queryURLReplace:
+      url:
+        - regex: '.+/movie/(.*)/(.*).+'
```
Comment (bnkai) [inline scrapers/AdultFilmIndex.yml:27]: althought it seems to work the regex seems a bit weird, eg
<https://regex101.com/r/LgAWOB/1>
if you only want to keep the movie id i would suggest something like this `.+/movie/([^/]+)/.+$`
instead <https://regex101.com/r/pLE432/1>

Comment (themese8) [reply scrapers/AdultFilmIndex.yml]: You are right, that's much better regex, updated the PR

**Comment by MichelFolco** [comment]
Cant download and save images anymore I get the error message when pressing save after scrapping:  processing front image: http error 418

**Pull request merged**